test/compiler/EscapeAnalysis/EAUtils.jl

```diff
@@ -176,9 +177,11 @@ function cache_escapes!(interp::EscapeAnalyzer,
 end
 
 function get_escape_cache(interp::EscapeAnalyzer)
-    return function (linfo::Union{InferenceResult,MethodInstance})
+    return function (linfo::Union{InferenceResult,MethodInstance,SemiConcreteResult})
         if isa(linfo, InferenceResult)
             ecache = get(interp.cache, linfo, nothing)
+        elseif isa(linfo, SemiConcreteResult)
+            ecache = get(interp.cache, linfo, nothing)
```

Comment:
This never returns cache (as `interp.cache` is `IdDict{InferenceResult,EscapeCache}` object). We should tweak callsites of `get_escape_cache` instead.